### PR TITLE
10 Build Static Version from Mockups

### PR DESCRIPTION
**Pull request by andela-uawili**
#10 

#### What does this PR do?
Creates a static (markup and styles only with hard-coded data) build of the web-client from the designed mockups. This static build - found in the `static_build` directory - serves to:
1. Demonstrate the intended behavior and interactions for the web-client components.
2. Provide the needed markup and styles when building out the React components.

#### Installation and Test Plan

1. Checkout to the `10_build_static_version_from_mockups` branch (at least before it's merged into staging) and cd into the `static_build` directory from the project root.
2.  Install the dependencies and start the static build server using:
```
$ npm install
$ npm start
```
3. You can now view the static page on running on `localhost:8081`.

#### Trello Card/Story:
https://trello.com/c/qZcKbDRP

### CODE REVIEW

**Comment by andela-kanyanwu** [comment]
@andela-rsadiq  please can you help review this alongside @andela-ashuaib

**Comment by andela-kanyanwu** [comment]
@andela-kadeniyi if you have some free time please help review also. Thanks.

**Comment by andela-kanyanwu** [comment]
@mfalade can you help review this? We need to get it out ASAP.

File: static_build/scss/components/_day_picker.scss [hunk 1]
```diff
@@ -0,0 +1,148 @@
+/* DayPicker styles */
+$day-picker-padding: 10%;
+$calendar-row-height: 40px;
+$calendar-day-size: 30px;
+
+.DayPicker {
```
Comment (andela-rsadiq) [inline static_build/scss/components/_day_picker.scss]: wondering why we are using capitalized-camel-cased form for our class name. is it a semantic ui thing? I would have expected `.dayPicker` or `.day-picker` not `.DayPicker`

File: static_build/scss/components/_day_picker.scss [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  justify-content: center;
+  position: relative;
+  max-width: 520px;
+  padding: 15px $day-picker-padding;
+  margin: auto;
+  user-select: none;
+  outline: none;
+}
+
+  .DayPicker-NavBar {
```
Comment (andela-rsadiq) [inline static_build/scss/components/_day_picker.scss]: same here with the className

File: static_build/scss/components/_day_picker.scss [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    align-items: center;
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
+    width: 100%;
+    height: $calendar-row-height;
+    padding: 0 $day-picker-padding;
+  }
+
+    .DayPicker-NavButton {
```
Comment (andela-rsadiq) [inline static_build/scss/components/_day_picker.scss]: same here with the className

File: static_build/scss/components/_day_picker.scss [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  cursor: default;
+  background-color: unset;
+}
+
+.DayPicker-Day--outside {
+  cursor: default;
+  color: $tvn-dark-grey;
+}
+
+.DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
```
Comment (andela-rsadiq) [inline static_build/scss/components/_day_picker.scss]: same here with the className. Please look through entire file `(_day_picker.scss)` if correction ought to be made?

File: static_build/scss/components/_generic_menu_bar_selector.scss [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    align-items: center;
+    width: 100%;
+    height: $tvn-dropdown-footer-height;
+    border-top: $tvn-stroke;
+    overflow: hidden;
+    @include tvn-theme-normal;
+  }
+
+  .dropdown.menu .list .item {
+    display: flex !important;
```
Comment (andela-rsadiq) [inline static_build/scss/components/_generic_menu_bar_selector.scss]: can we explore CSS specificity here as against `!important`?

File: static_build/scss/components/_generic_menu_bar_selector.scss [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+      margin: 0;
+      width: 100%;
+      white-space: nowrap;
+      overflow: hidden;
+      line-height: 1.3em;
+    }
+    .description {
+      margin-top: 3px;
+      color: $tvn-black;
+      font-size: $tvn-font-size-small !important;
```
Comment (andela-rsadiq) [inline static_build/scss/components/_generic_menu_bar_selector.scss]: same here with CSS specificity as against `!important`

File: static_build/scss/components/_serving_card.scss [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+  }
+
+  &:hover {
+    border: $tvn-accented-soft-stroke;
+    //.image.wrapper .image {
+    //  transform: scale(1.05);
+    //}
```
Comment (andela-rsadiq) [inline static_build/scss/components/_serving_card.scss]: can we take out unused lines of code?

**Comment by andela-rsadiq** [comment]
@andela-uawili please find comments written above and correct them as appropriate. Cheers!

File: static_build/css/style.css [hunk 8]
```diff
@@ -0,0 +1,988 @@
+/* Mobile */
+@media only screen and (max-width: 767px) {
```
Comment (andela-uawili) [inline static_build/css/style.css]: As you noted below, this is already being done. Thanks

File: static_build/scss/components/_day_picker.scss (hunk 1 above)
Comment (andela-uawili) [inline static_build/scss/components/_day_picker.scss]: @andela-rsadiq For the calendar widget, I opted to use the popular [react-day-picker](http://react-day-picker.js.org/) component. To apply our custom styles to the component, the [styling section in the docs](http://react-day-picker.js.org/Styling.html) mandates that we use these predefined classes so as much as we'd love to define our convention, we have to stick with theirs for that component.

Their class naming convention loosely follow a widely used CSS methodology called BEM. You can read more about it [here](https://css-tricks.com/bem-101/)

File: static_build/scss/components/_day_picker.scss (hunk 3 above)
Comment (andela-uawili) [inline static_build/scss/components/_day_picker.scss]: Please, see my comment above.

File: static_build/scss/components/_day_picker.scss (hunk 4 above)
Comment (andela-uawili) [inline static_build/scss/components/_day_picker.scss]: Please, see my comment above.

File: static_build/scss/components/_generic_menu_bar_selector.scss (hunk 5 above)
Comment (andela-uawili) [inline static_build/scss/components/_generic_menu_bar_selector.scss]: @andela-rsadiq For the most part, I relied on specificity and avoided using`!important` wherever I could. It is rather unfortunate that Semantic UI applies this 'abomination' in some of their style rules which I needed to override with our custom styles.
Given that the only way to override a `!important` is with another `!important`, I was given no other choice but to succumb (reluctantly though) to using this $%#@#$% in ours.

My earnest hope is that my unwilling ethical sacrifice will save others the horrors of doing the same. Long live CSS!!! Long Live Taverna!!!

PS: Please forgive the dramatics, just thought I could spice up the with a little humor ;)

File: static_build/scss/components/_serving_card.scss (hunk 7 above)
Comment (andela-uawili) [inline static_build/scss/components/_serving_card.scss]: Yes, this should go.

File: static_build/scss/components/_generic_menu_bar_selector.scss (hunk 6 above)
Comment (andela-uawili) [inline static_build/scss/components/_generic_menu_bar_selector.scss]: Please, see my `!important` comment above.

File: static_build/scss/_global.scss [hunk 9]
```diff
@@ -0,0 +1,266 @@
+
+// RESET AND SEMANTIC UI OVERRIDES:
+// ===================================================================================
+body {
```
Comment (andela-uawili) [inline static_build/scss/_global.scss]: :)

**Comment by andela-uawili** [comment]
@andela-rsadiq please re-review.

File: static_build/scss/components/_serving_card.scss (hunk 7 above)
Comment (andela-uawili) [inline static_build/scss/components/_serving_card.scss]: ... gone!

File: static_build/scss/components/_day_picker.scss (hunk 1 above)
Comment (andela-rsadiq) [inline static_build/scss/components/_day_picker.scss]: gotcha 👍

File: static_build/scss/components/_generic_menu_bar_selector.scss (hunk 6 above)
Comment (andela-rsadiq) [inline static_build/scss/components/_generic_menu_bar_selector.scss]: nice one😂

**Pull request merged**